### PR TITLE
simplified dhtfullsync2 && Busychat test

### PR DESCRIPTION
**Pull request by neonphog**
- fixes many entry and metadata syncing bugs in realmode
- simplifies dht-fullsync messaging strategy
- busychat test app

also see https://github.com/holochain/holochain-rust/pull/1379

### CODE REVIEW

File: tools/busychat/index.js [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+      case 'fetchEntryResult':
+        if (!this._channels.has(e.dnaAddress)) {
+          return
+        }
+
+        this._display(
+          '[' + e.dnaAddress + ']', e.content.from + ':', e.content.data)
+        break
+      case 'fetchMetaResult':
+        // console.log(e)
```
Comment (ddd-mtl) [inline tools/busychat/index.js]: Maybe use meta as just a 'reply to' link to another entry.

Comment (neonphog) [reply tools/busychat/index.js]: Oo, I like that idea. Yeah, it's kind of a fun challenge to implement something that is functional while also mimicing the patterns of holochain

File: lib/hackmode/config.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ module.exports = exports = config.createDefinition({
         await mosodium.hash.sha256(Buffer.from(base.id), tmp)
         base.idHash = tmp.toString('base64')
       }
+    }),
+    transientTransportId: config.entry({
```
[the remark this answers is not in the document]
Comment (neonphog) [reply lib/hackmode/config.js]: good call :+1: - done

File: tools/busychat/index.js [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+        this._stats.entryCount += 1
+        this._updatePrompt()
+
+        this._fetchMessage(e.dnaAddress, e.address)
+
+        // if (!this._channels.has(e.dnaAddress)) {
+        //   return
+        // }
+        // this._display(
+        //   '[' + e.dnaAddress + ']', e.content.from + ':', e.content.data)
```
[the remark this answers is not in the document]
Comment (neonphog) [reply tools/busychat/index.js]: removed :+1:

**Pull request merged**